packages/components/date-input/template.njk

```diff
@@ -63,7 +66,7 @@
         },
         id: item.id if item.id else (params.id + "-" + item.name),
         classes: "nhsuk-date-input__input " + (item.classes if item.classes),
-        name: (params.namePrefix + "-" + item.name) if params.namePrefix else item.name,
+        name: (params.namePrefix + "[" + item.name + "]") if params.namePrefix else item.name,
         value: item.value,
```

Comment:
```suggestion
        value: item.value or values[item.name],
```

Untested idea. At the moment `values` only gets used if you've *not* provided items. Should that be the case? We can default to directly assigned value, or else fall back to values.

Reply:
Ooh interesting idea. I'd not considered that it might also be beneficial where you’ve provided custom items as well as just for the default day, month, year one.

I think this should work though - and would mean that even if you did custom items (say only month and year), then you'd still only have to assign `values` once at the top level?

Reply:
>I think this should work though - and would mean that even if you did custom items (say only month and year), then you'd still only have to assign values once at the top level?

Yes that was my thinking - that `items` and `values` doesn't have to be mutually exclusive - and perhaps is confusing if it were. We'd need to test it works though.

Reply:
This works - just needed to make one change from `values[item.name]` to `params.values[item.name]`.

Committed in 8a703057b4ed3bbf64fabd2d55dcb5296bcc1aa7